src/unwrap.jl

```diff
@@ -0,0 +1,370 @@
+module Unwrap
+
+using Compat: selectdim
+
+export unwrap, unwrap!
+
+"""
+    unwrap!(m; range=2pi, wrap_around=tuple(fill(false, N)...), seed=-1)
+
+In-place version of unwrap(m; range, wrap_around, seed)
+"""
+unwrap!
+
+"""
+    unwrap(m::AbstractArray{T, N}; range=2pi, wrap_around=tuple(fill(false, N)...), seed=-1)
+
+Assumes `m` to be an array of values of arbitrary dimension that
+have been wrapped to be inside the given range (centered around
+zero), and undoes the wrapping by identifying discontinuities. To operate on a
+subset of dimensions of an array, simply pass in the corresponding `view`.
+
+A common usage is for a phase measurement over time, such as when
+comparing successive frames of a short-time-fourier-transform, as
+each frame is wrapped to stay within (-pi, pi].
+
+# Arguments
+- `A::AbstractArray{T, N}`: Array to unwrap
+- `range::Number`: Range of wrapped array
+- `wrap_around::NTuple{N, Bool}`:  When an element of this tuple is `true`, the
+    unwrapping process will consider the edges along the corresponding axis
+    of the image to be connected
+- `seed::Int`: Unwrapping of arrays with dimension > 1 uses a random initialization. This
+    sets the seed of the RNG.
+"""
+unwrap
+
+function unwrap(m::AbstractArray{T, N}, args...; kwargs...) where {T<:AbstractFloat, N}
+    unwrap!(copy(m), args...; kwargs...)
+end
+
+### Deprecated unwrap function
+function unwrap!(m::Array{T}, dim::Integer; range::Number=2pi) where T<:AbstractFloat
+    Base.depwarn(string("`unwrap!(m::Array{T}, dim::Integer; range::Number=2pi) ",
+        " where T<:AbstractFloat` is deprecated, ",
+        "use `unwrap!(m; range)` instead if your data has more than one dimension."),
```

Comment:
This doesn't tell me how I get the old behavior that I'd need if I have e.g. multiple one-dimensional signals collected in a matrix that I'd like to individually unwrap.